tainting: Set size limits to prevent memory blowups

IagoAbal opened the pull request:
Follows: c5640661b82 ("tainting: pro: JS/TS: Interprocedural field-sensitivity (#7867)")

test plan:
See returntocorp/semgrep-proprietary#750

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

emjin commented:
I guess failing tests means that the limits are too small?

emjin commented:
Can you also open the semgrep-proprietary PR to update develop so that we can see the effect there?

IagoAbal committed:
tainting: Set size limits to prevent memory blowups

Follows: c5640661b82 ("tainting: pro: JS/TS: Interprocedural field-sensitivity (#7867)")

test plan:
make test

aryx commented on src/tainting/Taint_lval_env.ml:
```diff
@@ -25,6 +25,24 @@ module VarMap = Var_env.VarMap
+ * not propagate taint for data with Boolean and integer type. Improving some
+ * of the data structures involved may help too.
+ *)
+let max_TAINTED_LVALS = 100
```
maybe worth putting in a separate file like src/configuring/Limits_semgrep.ml

IagoAbal committed:
Yoann's review

IagoAbal committed:
Fix changelog
